src/adamrenklint/boot_exec.clj

```diff
@@ -13,5 +14,6 @@
   (let [pred? (if pred-fn (resolve pred-fn) (fn [] true))]
     (core/with-pass-thru _
       (when (pred?)
-        (binding [util/*sh-dir* (or dir ".")]
-          (apply util/dosh (string/split cmd #" ")))))))
+        (binding [util/*sh-dir* (or dir ".")
+                  os-cmd (if windows? (str "cmd /c " cmd) cmd)]
+          (apply util/dosh (string/split os-cmd #" ")))))))
```

Comment:
I'm almost sure this is not the most clean and idiomatic 3 lines of cljs, so please be merciless in your review.

Reply:
The `binding` macro is used to redefine an existing dynamic var in code that you don't control, and should really only be used on vars that are defined with the `^:dynamic` flag and has a asterisk-wrapped name, like `*sh-dir*.

In this case, you only really need to create a local binding, which is done with the `let` macro. So I'd write this code like this:

```clojure
(binding [util/*sh-dir* (or dir ".")]
  (let [os-cmd (if windows? (str "cmd /c " cmd) cmd)]
    (apply util/dosh (string/split os-cmd #" "))))
```

## Links

- https://cljs.github.io/api/cljs.core/binding
- https://cljs.github.io/api/cljs.core/let

Reply:
Thanks for the tip! I've made it the way you suggested.